### PR TITLE
Output junit xml from smoketest, examples and chromatic

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: https://linear.app/chromaui/issue/SB-640/generate-junitxml-for-current-e2e-tests-config-to-get-better-error

## What I did

- [x] Added junit xml results from our `smoke-test` step
- [x] Added junit xml results from our `examples` step
- [x] Add junit xml results from our `chromatic` step

image

image

## What I didn't do

- Add it to the test runner (we don't currently use the test runner, I will add when we do)
- Add it to cypress (I already have it in the playwright PoC: #18826) 

## How to test

See CI job

### CODE REVIEW

**Comment by tmeasday** [comment]
I'll do another pass over this code when we setup the scripts for the new circle jobs (convert to TS, generalize, maybe use a different junit lib)

File: scripts/smoketest-storybooks.js [hunk 1]
```diff
@@ -118,7 +111,48 @@ const run = async () => {
 
   if (deployables.length) {
     logger.log(`🏗  Will smoke test: ${chalk.cyan(deployables.join(', '))}`);
-    await handleExamples(deployables);
+    const s = new Date();
```
Comment (yannbf) [inline scripts/smoketest-storybooks.js:114]: there's probably a better variable name for this? :p

File: scripts/build-storybooks.js [hunk 2]
```diff
@@ -202,7 +195,49 @@ const run = async () => {
...
+              systemOut: output,
+              ...(!ok && {
+                errors: [err],
+              }),
+            },
+          ],
+        })),
+      });
+      await outputFile(junitPath, junitXml);
+      console.log(`Test results written to ${resolve(junitPath)}`);
```
Comment (yannbf) [inline scripts/build-storybooks.js:224]: console.log vs logger.log?

**Review by yannbf — APPROVED** [review]

**Comment by tmeasday** [comment]
Closing in favour of https://github.com/storybookjs/storybook/pull/18922

**Pull request closed without merging**